### ISSUE TITLE
zooming in on 3D plot to within axes causes object to be rendered outside the axes

### ISSUE THREAD

**Issue by bjarthur** [MakieOrg/Makie.jl#3584]
- [x] are you running newest version (version from docs) ?
- [x] can you reproduce the bug with a fresh environment ? (`]activate --temp; add Makie`)
- [x] What platform + GPU are you on?

![test](https://github.com/MakieOrg/Makie.jl/assets/1538268/9f978166-5fb6-489a-b710-33e47e2e8536)

```
julia> using GLMakie

julia> f,a,s = scatter(Point3f.([0,1],[0,1],[0,1]))

julia> fps = 10
10

julia> record(f.scene, "test.gif"; framerate = fps) do io
           for i = 1:200
               sleep(1/fps)
               recordframe!(io)
           end
       end
"test.gif"

julia> versioninfo()
Julia Version 1.10.0
Commit 3120989f39b (2023-12-25 18:01 UTC)
Build Info:
  Official https://julialang.org/ release
Platform Info:
  OS: macOS (arm64-apple-darwin22.4.0)
  CPU: 12 × Apple M2 Max
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-15.0.7 (ORCJIT, apple-m1)
  Threads: 1 on 8 virtual cores
Environment:
  JULIA_PROJECT = @.
  JULIA_EDITOR = vi

(jl_ZxNi4Q) pkg> st
Status `/private/var/folders/s5/8d629n5d7nsf37f60_91wzr40000gq/T/jl_ZxNi4Q/Project.toml`
  [e9467ef8] GLMakie v0.9.6
```

**Comment by ffreyer** [comment]
This and #3544 seem to be an issue with `(projectionview * linepoint)[4]` becoming negative. We could fix this by just setting `gl_Position.w = max(1e-4, gl_Position.w)` in shaders, but I would assume there is a deeper reason for this which we should fix. I thought maybe it's a near/far value problem, but those are fine (e.g. 0.06 and 5.5). `projectionview[4][4]` becomes negative a bit after the problem starts happening so I think that's fine too.

**Comment by ffreyer** [comment]
Fixed by #3651 via #3650 in 0.21
